### PR TITLE
fix : Added flag and updated doc to verify worker readiness only if cluster is accessible.

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description
This change is done to create a flag so as to validate the network health check only when cluster is accessible and update the doc too with this information.

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [x] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -284,10 +296,18 @@ resource "ibm_container_vpc_worker_pool" "autoscaling_pool" {
...
+
+# Please note - This network check is only applicable in case of public
+# endpoints and will not be checked if private endpoint is used.
+
 ##############################################################################
 
 resource "null_resource" "confirm_network_healthy" {
 
+  # If private endpoint then do not create this resource.
+  count = var.disable_public_endpoint ? 0 : 1
```
Comment (ocofaigh) [inline main.tf:308]: hmm, didn't we say that the consumer may have access to their private endpoint, and so may still wish to run this script. Instead we should probably have a variable called `confirm_network_healthy` and default it to `true`, however in the variable description make sure it mentions that in order to set to true, the runtime must have access to kube

File: variables.tf [hunk 2]
```diff
@@ -154,4 +154,10 @@ variable "vpc_id" {
   description = "Id of the VPC instance where this cluster will be provisioned"
 }
 
-#############################################################################
+variable "confirm_network_healthy" {
+  type        = bool
+  description = "Set the flag as true only if runtime is able to access kubernetes cluster otherwise change the value to false - Default : true"
```
Comment (ocofaigh) [inline variables.tf:159]: Update the description to:

By setting this to true, a script will run kubectl commands to verify that all worker nodes can communicate successfully with the master. If the runtime does not have access to the kube cluster to run kubectl commands, this should be set to false

File: variables.tf [hunk 3]
```diff
@@ -154,4 +154,10 @@ variable "vpc_id" {
   description = "Id of the VPC instance where this cluster will be provisioned"
 }
 
-#############################################################################
+variable "confirm_network_healthy" {
```
Comment (ocofaigh) [inline variables.tf:157]: perhaps go with the variable name `verify_worker_network_readiness`

Comment (imprateeksh) [reply variables.tf:157]: Noted. will make the change.

File: variables.tf (hunk 2 above)
Comment (imprateeksh) [reply variables.tf:159]: Sure. Thank you.

**Comment by ocofaigh** [comment]
Please address the failing pre-commit hook

**Review by ocofaigh — APPROVED** [review]

**Comment by ocofaigh** [comment]
@imprateeksh I have created an internal issue and assigned to you to address https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/pull/68#issuecomment-1479910490

**Comment by imprateeksh** [comment]
ok @ocofaigh. I will start looking into it.

**Comment by imprateeksh** [comment]
> 

Sure @vburckhardt - I will work on creating a separate example. I remember doing the same thing which you suggested as   "we also add a count = var.verify_worker_network_readiness ? 1 : 0 on that data block"  and landing into some error related to `ibm_container_vpc_worker_pool`

But I will try the changes once again by adding example and seeing if I am getting same error again.

**Pull request merged**